### ISSUE TITLE
[Core]: Implement missing accessibility functionality in carousel

### ISSUE THREAD

**Issue by davidmenendez** [carbon-design-system/carbon-ai-chat#1025]
proxy issue to track the [core issue](https://github.com/carbon-design-system/carbon/issues/21680)

## acceptance criteria

- [x] include basic missing aria labels
  - `aria-live`
  - `aria-atomic`
  - `aria-hidden`
- [x] include missing helper regions
  - `liveregion visuallyhidden`
- [x] verify proper functionality and aria labels implemented for slide transitions

### resources

* [carousels by W3C](https://www.w3.org/WAI/tutorials/carousels/full-code/)
